xbmc/video/VideoDatabase.cpp

```diff
@@ -3964,22 +3964,39 @@ void CVideoDatabase::GetSameVideoItems(CFileItem& item, CFileItemList& items)
 
     m_pDS->close();
 
-    // get items with same title
+    // get items with same title and year
     std::string title = item.GetVideoInfoTag()->GetTitle();
+    int year = item.GetVideoInfoTag()->GetYear();
+    bool hasYear = item.GetVideoInfoTag()->HasYear();
 
     VideoDbContentType itemType = item.GetVideoContentType();
     if (itemType == VideoDbContentType::MOVIES)
     {
+      std::vector<int> ids;
       m_pDS->query(PrepareSQL("SELECT idMovie FROM movie WHERE movie.c%02d = '%s'",
                               VIDEODB_ID_TITLE, title.c_str()));
 
       while (!m_pDS->eof())
       {
-        insertItemId(m_pDS->fv("idMovie").get_asInt());
+        ids.push_back(m_pDS->fv("idMovie").get_asInt());
         m_pDS->next();
       }
 
       m_pDS->close();
+
+      for (const auto id : ids)
+      {
+        CFileItem current;
+        if (GetDetailsByTypeAndId(current, itemType, id))
```

Comment:
I am a little concerned about all the additional DB queries that are done in the background here and the performance impact they might have on the scraping process, especially when you do an initial import of your library.

Reply:
Don't need a join.  You could just do `SELECT idMovie, substring(Premiered,1,4) as year FROM movie WHERE movie..c%02d = '%s'", VIDEODB_ID_TITLE, title.c_str()));`

AFAIK, it is preferable to fetch as much of the needed info from the db with as few calls as possible.

Reply:
> Good points, yes there is an impact for external databases, although in this case there probably wouldn't be a lot of matches for one movie name and therefore not many extra requests. But still...

if you scan a library with thousands of movies, queries add up. And our scraping aready isn't the fastest anyways

Reply:
The movie table only has a premiered date. 

https://github.com/xbmc/xbmc/blob/master/xbmc/video/VideoDatabase.cpp#L4285

You can do it all in the sql query.  `select idmovie ,case when instr('-',movie.premiered) > 5 then substring(premiered,7,4) else substring(premiered,1,4) end as year from movie where movie.c%02d = '%s'", VIDEODB_ID_TITLE, title.c_str()));`  

I think that will handle yyyy-mm-dd, mm-dd-yyyy and just yyyy for both MySQL/mariadb & sqlite

Reply:
just a thought, but couldn't we already check for a matching year in the query? So either one complex query covering all cases (incoming movies has year or not), or a dedicated query for `incoming movie has year` and `incoming movies has no year`. That way we eliminate all the other overhead

Reply:
For library scanning, the time-consuming part should be file system overhead and downloading from Internet, is it really necessary to go with above ways?

Reply:
performance overhead certainly matters, especially for low powered devices like RPI and other SBCs. And if you have a look at `GetDetailsByTypeAndId` it will at least trigger two or three additional queries (especially since this method will also trigger `HasVideoVersions()`, for which we already have a complaint that this feature noticeably slows down the listing of larger movie libraries which we also have to address)

Reply:
I'm not saying the perf overhead is not matter, I'm saying in comparison with the big factors of perf, is this really necessary? The previous complaint about perf was because the old code tried to query DB for a second time during media window update, now I cached this information in VideoInfoTag, and didn't see any perf complaint currently. Anyway, I can cache the GetDetailsByTypeAndId result as you explained in OP as a solution.